Documentation/api.md

```diff
@@ -277,8 +277,12 @@ PodMetricsEndpoint defines a scrapeable endpoint of a Kubernetes Pod serving Pro
 | params | Optional HTTP URL parameters | map[string][]string | false |
 | interval | Interval at which metrics should be scraped | string | false |
 | scrapeTimeout | Timeout after which the scrape is ended | string | false |
+| tlsConfig | TLS configuration to use when scraping the endpoint. | *[TLSConfig](#tlsconfig) | false |
```

Comment:
I would prefer if we didn't re-use the exact TLSConfig struct here, for the same reason why we don't want to allow setting arbitrary `bearerTokenFile` values. To not repeat ourselves too much what we could do is extract the "safe" TLS configuration bits into a separate struct and embed that struct into the existing TLSConfig struct as well as into the new "PodMonitorEndpointTLSConfig" struct.

Reply:
I created a new type SafeTLSConfig and moved all the safe configuration options into it.
The TLSConfig structure now contains an embedded SafeTLSConfig object and its legacy unsafe options.